api.md

```diff
@@ -0,0 +1,242 @@
+# GOSwap Stats API
+
+TODO finish documenting descriptions
+TODO document all fields
+TODO make the actual API look like this (WARNING: it currently does not)
+
+### list tokens
+
+`/v1/tokens`
+
+`
+{
+  "tokens": [
+    {
+      "name": "string",
+      "symbol": "string",
+      "decimals": 123,
+      "address": "0xstring"
+    }
+  ]
+}
+`
+
+### get token
+
+`/v1/tokens/{address}`
+
+`
+{
+  "token": {
+    "name": "string",
+    "symbol": "string",
+    "decimals": 123,
+    "address": "0xstring"
+  }
+}
+`
+
+### list pairs
+
+`/v1/pairs`
+
+`
+{
+  "pairs": [
+    {
+      "index": 123,
+      "pair": "STRING-STRING",
+      "address": "0xstring",
+      "token0": "0xstring",
+      "token1": "0xstring"
+    }
+  ]
+}
+`
+
+### get pair
+
+`/v1/pairs/{address}`
+
+`
+{
+  "pair": {
+    "index": 123,
+    "pair": "STRING-STRING",
+    "address": "0xstring",
+    "token0": "0xstring",
+    "token1": "0xstring"
+  }
+}
+`
+
+### list stats totals
+
+Returns a sum of stat totals across all tokens/pairs that are `time_frame`
+apart, between `time_start` and `time_end`.
+
+TODO defaults for `time_` fields
+TODO do we want string encapsulated floats?
```

Comment:
any thoughts on this @treeder ? it's a bit odd, and I think it's simply for decimal.Decimal package (ie it's the default), which notes things about javascript having issues losing precision when parsing them. I'm not sure if this is intentional ("a feature"), as I simply don't have any context for why we didn't use floats vs the decimal package (I do see the firebase oddness, unsure if related).

Reply:
If we care about precision, then yes, we need to stringify them. If we don't really care, then it could be floats. For stats, we probably don't care 100%. But I could foresee that it might matter in certain cases, so maybe best to just use strings across the board so we don't end up with some floats and some strings.